### PR TITLE
feat: renew route lrucache when the routes change

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Use the fix from #8119 and resolve the lrucache and router match loops.

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/ai.lua [hunk 1]
```diff
@@ -280,7 +281,8 @@ end
 
 
 function _M.init_worker()
-    orig_router_match = router.router_http.match
+    orig_router_match = base_router.match_uri
```
Comment (membphis) [inline apisix/plugins/ai.lua:284]: we need a better name for `orig_router_match`

File: apisix/plugins/ai.lua [hunk 2]
```diff
@@ -70,21 +67,20 @@ local orig_http_balancer_phase = apisix.http_balancer_phase
 
 local default_keepalive_pool = {}
 
-local function match_route(ctx)
-    orig_router_match(ctx)
-    return ctx.matched_route or false
+local function match_route(uri_router, match_opts, api_ctx)
```
Comment (membphis) [inline apisix/plugins/ai.lua:70]: we need a better name: `match_route `

**Review by membphis — COMMENTED** [review]
name style

File: apisix/plugins/ai.lua [hunk 3]
```diff
@@ -224,15 +220,20 @@ local function routes_analyze(routes)
          or route_flags["service_id"]
          or route_flags["plugin_config_id"]
          or global_rules_flag then
-        router.router_http.match = orig_router_match
+            base_router.match_uri = orig_router_match
```
Comment (membphis) [inline apisix/plugins/ai.lua:223]: bad indentation

File: apisix/plugins/ai.lua [hunk 4]
```diff
@@ -224,15 +220,20 @@ local function routes_analyze(routes)
...
     else
         core.log.info("use ai plane to match route")
-        router.router_http.match = ai_match
+        base_router.match_uri = ai_match_uri
+        local count = #routes + 3000
```
Comment (membphis) [inline apisix/plugins/ai.lua:227]: pls add one more blank line before this line

**Review by membphis — COMMENTED** [review]
code style

File: apisix/plugins/ai.lua (hunk 3 above)
Comment (tzssangglass) [reply apisix/plugins/ai.lua]: done

File: apisix/plugins/ai.lua (hunk 2 above)
Comment (tzssangglass) [reply apisix/plugins/ai.lua]: done

File: apisix/plugins/ai.lua (hunk 1 above)
Comment (tzssangglass) [reply apisix/plugins/ai.lua]: done

**Review by membphis — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**